api/src/main/java/org/ccci/idm/user/Dn.java

```diff
@@ -112,6 +125,15 @@ public Component(@Nonnull final String type, @Nonnull final String value) {
         }
 
         @Override
+        public int compareTo(@Nonnull final Component o) {
+            int resp = value.compareToIgnoreCase(o.value);
+            if (resp == 0) {
+                resp = type.compareToIgnoreCase(o.type);
+            }
+            return resp;
+        }
```

Comment:
true, I could have inlined part of it using the ternary operator (also know as the elvis operator `?:`)

I'll leave it as is, someone can refactor into that if they feel more drawn to it in this use case :)